packages/db/fauna/resources/Function/renameUserUpload.js

```diff
@@ -0,0 +1,66 @@
+import fauna from 'faunadb'
+
+const {
+  Abort,
+  Collection,
+  CreateFunction,
+  Exists,
+  Function,
+  Get,
+  If,
+  Index,
+  IsNonEmpty,
+  Lambda,
+  Let,
+  Match,
+  Query,
+  Ref,
+  Select,
+  Update,
+  Var
+} = fauna
+
+const name = 'renameUserUpload'
+const body = Query(
+  Lambda(
+    ['userId', 'cid', 'name'],
+    Let(
+      {
+        userRef: Ref(Collection('User'), Var('userId')),
+        contentMatch: Match(Index('unique_Content_cid'), Var('cid'))
+      },
+      If(
+        IsNonEmpty(Var('contentMatch')),
+        Let(
+          {
+            content: Get(Var('contentMatch'))
+          },
+          Let(
+            {
+              uploadMatch: Match(
+                Index('upload_by_user_and_content'),
+                Var('userRef'),
+                Select('ref', Var('content')),
+                true
+              )
+            },
+            If(IsNonEmpty(Var('uploadMatch')),
+              Update(
```

Comment:
Check for non-empty name?

Reply:
hummm, shouldnt' upload name be required instead? I thought it was, but just confirmed it is not required in the upload collection schema

Reply:
I think that's ok. I was just thinking about the UX of seeing blank as the upload name, but I guess it could show "No Name" or something.

I think this is not required. Folks may want to remove the name...lets not stop them from doing that.